src/main/java/net/fabricmc/mapping/util/StringPool.java

```diff
@@ -0,0 +1,39 @@
+package net.fabricmc.mapping.util;
+
+import java.util.HashMap;
+
+/**
+ * A pool of strings used for de-duplication.
+ */
+public class StringPool {
+    // The HashMap uses value-equality to de-duplicate keys. Because we need a value type, we simply stuff the key
+    // reference back in to make the code below pretty.
+    private final HashMap<String, String> map = new HashMap<>();
+
+    /**
+     * Adds a string to the pool, returning a de-duplicated reference to the string.
+     * @param str The string to de-duplicate
+     * @return A de-duplicated reference to a string with the same contents as {@param str}
+     */
+    public String pool(String str) {
+        return this.map.computeIfAbsent(str, StringPool::identity);
```

Comment:
Function.identity()?

Reply:
Nevermind, that creates a new instance every call

Reply:
> Nevermind, that creates a new instance every call

As far as I know it's not specified, the JLS explicitly does not guarantee anything about reference equality of lambdas. In practice it seems to return the same instance on every call (and I would be surprised if it didn't):
```java
import java.util.function.Function;

public class Test {
    public static void main(String[] args) {
        Function<String, String> idA = Function.identity();
        Function<String, String> idB = Function.identity();
        System.out.println(idA == idB);
    }
}
```
prints `true` for me, on AdoptOpenJDK Hotspot builds for Java 8, 14 and 15. I would use `Function.identity()` and trust the JVM, but @jellysquid3 probably knows these things much better than I do.

Reply:
The only reason the `identity` method was created was because I forgot about `Function.identity()`. They're functionally the same in bytecode.